docs/DeveloperGuide.md

```diff
@@ -173,19 +173,36 @@ The class also has getters and setters for the fields, as well as a `toString()`
 
 #### Current Implementation
 
-It is currently implemented as a `DashboardCommand` that is executed by the `LogicManager` when the user enters the `dashboard` command. This command returns a `CommandResult` object that contains the following statistics:
-- Total number of interactions across all persons.
-- Percentage of person with `INTERESTED` outcome.
-- Percentage of person with `NOT_INTERESTED` outcome.
+The currently active view is managed by the `MainWindow` class. Depending on the `showDashboard` flag of the last `CommandResult`, the `MainWindow` will display either the `DashboardDisplay` or `ClientDisplay` object. By default, i.e. on launch, the dashboard view is shown first.
+
+Statistics shown on the dashboard currently include the following:
+- Breakdown of clients by interaction outcome
+  - `Uncontacted`: clients that have no interactions yet
+  - `Contacting`: clients that have at least one non-closing interaction (outcome is not `CLOSED`)
+  - `Closed`: clients that have at least one closing interaction (outcome is `CLOSED`)
+- Average number of interactions per client
+- Proportion of interactions by outcome
+- Number of clients by lead category (hot, warm, cold)
+
+The dashboard also shows the upcoming follow-ups with clients. The follow-ups are separated by date, sorted from earliest to latest. The follow-up date is calculated using the client's `lead` and the date of the latest interaction, see [Reminders](#Reminders) for more details.
+
+All data for statistics and follow-ups are retrieved from the `Dashboard` component.
 
 #### Design Considerations
 
+Currently, the dashboard only shows static data at a specific state in time. To achieve perceived dynamicism, any command that modifies the client list will switch the view to the client view. This is managed using the `showDashboard` flag mentioned previously. This will only work if there is only one instance of Connectify running at any time. If the data is modified from another instance of Connectify, the dashboard view will not be updated in real-time.
+
+In order to achieve real-time updates, we will need to implement a mechanism to track changes to the client data, and trigger the dashboard to update accordingly. However, this leads to another problem, which is the performance of the dashboard.
+
+As the amount of client data increases, the calculation of statistics and follow-ups may increase significantly. Therefore, the dashboard should be updated, and the statistics and reminders data recalculated, only when the dashboard view is actually shown, and data has been modified.
+
+The `Dashboard` component already has an `isDashboardDirty` flag to indicate if the dashboard data needs to be recalculated. However, due to time constraint, it is currently always set to `true` and not being used to track data changes to the client data. In the future, we can track changes to this flag, e.g. using observers, and update the dashboard accordingly.
```

Comment:
Should we add a future thingy section instead here or isit alright to leave the description of how it should be in the future like this?